src/java/com/sarality/app/view/dialog/CustomViewDialogFragment.java

```diff
@@ -85,6 +85,8 @@ protected void reinitialize() {
    * @param loaderId Id to used for the loader
    */
   protected <D> void startLoad(DataSource<D> dataSource, int loaderId, DataConsumer<D> consumer) {
+    // Kill all previously loaded data by the activity
+    getActivity().getSupportLoaderManager().destroyLoader(loaderId);
```

Comment:
OK, I see the problem now. Looks like it is really hard for the Action to figure out before hand what loaderId it should pass. 

Let me think about how we can solve that. We may need to start passing around a LoaderIdProvider and then each of the Loaders registers an enum with the Provider to make sure there are no conflicts.